packages/cardpay-cli/utils.ts

```diff
@@ -3,39 +3,84 @@ import Web3 from 'web3';
 import { AbstractProvider } from 'web3-core';
 import { HttpProvider, getConstant, networkIds, getConstantByNetwork } from '@cardstack/cardpay-sdk';
 import WalletConnectProvider from '@walletconnect/web3-provider';
-import { Options } from 'yargs';
+import { Options, Arguments } from 'yargs';
+/* eslint-disable @typescript-eslint/no-require-imports,@typescript-eslint/no-var-requires */
+const TrezorWalletProvider = require('trezor-cli-wallet-provider');
 
 const BRIDGE = 'https://safe-walletconnect.gnosis.io/';
 
-export async function getWeb3(network: string, mnemonic?: string): Promise<Web3> {
-  if (mnemonic) {
-    return new Web3(
-      new HDWalletProvider({
-        chainId: networkIds[network],
-        mnemonic: {
-          phrase: mnemonic,
+export type ConnectionType = 'mnemonic' | 'trezor' | 'wallet-connect';
+
+interface Web3OptsMnemonic {
+  connectionType: 'mnemonic';
+  mnemonic: string;
+}
+
```

Comment:
@habdelra `connectionType` is the main key here. I struggled a bit because some other data are required when using a different connectionType. In this case, the "mnemonic" has to be passed in